extensions/datasource/runtime/src/main/java/io/quarkus/datasource/runtime/DataSourceRuntimeConfig.java

```diff
@@ -15,15 +16,11 @@ public interface DataSourceRuntimeConfig {
      *
      * See xref:datasource.adoc#datasource-active[this section of the documentation].
      *
-     * If the datasource is not active, it won't start with the application,
-     * and accessing the corresponding Datasource CDI bean will fail,
-     * meaning in particular that consumers of this datasource
-     * (e.g. Hibernate ORM persistence units) will fail to start unless they are inactive too.
-     *
      * @asciidoclet
      */
     @WithDefault("true")
-    boolean active();
+    @ConfigDocDefault("`true` if the URL is set, `false` otherwise")
+    Optional<Boolean> active();
```

Comment:
If there is a `WithDefault`, it doesn't make sense to use `Optional`, good catch!

Reply:
Keeping the default=true was an oversight.

I wanted to use `Optional<Boolean>` so that we can detect misuse; for example in Hibernate ORM you can set `quarkus.hibernate-orm.enabled=false` and `quarkus.hibernate-orm.active=true`, both explicitly, and it will result in an error because, well, that doesn't make any sense.

Here, I'm wondering if we should raise an error when setting `quarkus.datasource.active=true` explicitly, but not setting a URL... Though you might want to make the datasource active, but only set the JDBC URL.

Oh, well, I'll just drop the optional, it's a problem for another day.

Reply:
> Oh, well, I'll just drop the optional, it's a problem for another day.

Reply:
> Oh, well, I'll just drop the optional, it's a problem for another day.

Actually, you could say the datasource isn't really active if the URL is not set, so... I'll drop `@WithDefault`, so that the next person using this config will at least ask themselves "what the hell does it mean when it's absent?".